src-tauri/src/controller_binaries.rs

```diff
@@ -484,3 +485,143 @@ pub async fn add_service(service: Service, state: State<'_, Arc<SharedState>>) -
     services_guard.insert(service.get_id()?, service);
     Ok(())
 }
+
+#[tauri::command(async)]
+pub async fn add_registry(
+    registry: Registry,
+    app_handle: AppHandle,
+    state: State<'_, Arc<SharedState>>,
+) -> Result<()> {
+    let store_path = app_handle
+        .path_resolver()
+        .app_data_dir()
+        .with_context(|| "Failed to resolve app data dir")?
+        .join("store.json");
+    let mut store = StoreBuilder::new(app_handle, store_path).build();
+    store.load().with_context(|| "Failed to load store")?;
+    if let Some(registries) = store.get("registries").cloned() {
+        match serde_json::from_value::<Vec<Registry>>(registries) {
+            Ok(mut registries) => {
+                registries.push(registry);
+                store
+                    .insert(
+                        "registries".to_string(),
+                        serde_json::to_value(&registries).unwrap(),
+                    )
+                    .with_context(|| "Failed to insert into store")?;
+                utils::fetch_all_services_manifests(&registries, &state)
+                    .await
+                    .expect("failed to fetch services")
+            }
+            Err(e) => println!("Error unwrapping registries: {:?}", e),
+        }
+    } else {
+        let new_registry = vec![registry.clone()];
+        store
+            .insert(
+                "registries".to_string(),
+                serde_json::to_value(&new_registry).unwrap(),
```

Comment:
```suggestion
        let new_registry = [&registry];
        store
            .insert(
                "registries".to_string(),
                serde_json::to_value(&new_registry).unwrap(),
```
no need to create a vector, both serialize to same Value.